cmd/switch.go

```diff
@@ -36,34 +40,77 @@ func newSwitchCmd(homedir string) *cobra.Command {
 		Example: `# switch the active istioctl version to version=1.7.4, flavor=tetrate and flavor-version=1
 $ getistio switch --version 1.7.4 --flavor tetrate --flavor-version=1`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			d := &api.IstioDistribution{
-				Version:       flagVersion,
-				Flavor:        flagFlavor,
-				FlavorVersion: int64(flagFlavorVersion),
+			d, err := switchParse(homedir, &flag)
+			if err != nil {
+				return err
 			}
-
 			return switchExec(homedir, d)
 		},
 	}
 
 	flags := cmd.Flags()
 	flags.SortFlags = false
-	flags.StringVarP(&flagVersion, "version", "", "", "Version of istioctl e.g. 1.7.4")
-	flags.StringVarP(&flagFlavor, "flavor", "", "", "Flavor of istioctl, e.g. \"tetrate\" or \"tetratefips\" or \"istio\"")
-	flags.IntVarP(&flagFlavorVersion, "flavor-version", "", -1, "Version of the flavor, e.g. 1")
-
-	_ = cmd.MarkFlagRequired("version")
-	_ = cmd.MarkFlagRequired("flavor")
-	_ = cmd.MarkFlagRequired("flavor-version")
+	flags.StringVarP(&flag.name, "name", "", "", "Name of istioctl, e.g. 1.9.0-istio-v0")
+	flags.StringVarP(&flag.version, "version", "", "", "Version of istioctl, e.g. 1.7.4")
+	flags.StringVarP(&flag.flavor, "flavor", "", "", "Flavor of istioctl, e.g. \"tetrate\" or \"tetratefips\" or \"istio\"")
+	flags.Int64VarP(&flag.flavorVersion, "flavor-version", "", -1, "Version of the flavor, e.g. 1")
 
 	return cmd
 }
 
+// if set name, it should only parse name to distro
+// if version, flavor and version are all set, just parse it to distro
+// if there exists active distro, switch with only one or two command will use the active distro setting for unset command
+// if there are no active distro exists, switch with only one or two command will use the default distro setting for unset command
+// if all commands are not set, use active setting if there has otherwise use default version
+// default version: latest version, default flavor: tetrate, default flavorversion: 0
+func switchParse(homedir string, flags *switchFlags) (*api.IstioDistribution, error) {
+	if len(flags.name) != 0 {
+		d, err := api.IstioDistributionFromString(flags.name)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse given name to %s istio distribution\n", flags.name)
+		}
+		return d, nil
+	}
+	fetched, err := manifest.FetchManifest()
+	if err != nil {
+		return nil, fmt.Errorf("cannot fetch istio manifest\n")
+	}
+
+	currDistro, _ := istioctl.GetCurrentExecutable(homedir)
```

Comment:
When would this return error? I see that this is not expected since we download the latest Istio at the download of GetIstio, and the current executable would never be empty normally.

Reply:
Then, we do not need to have ` latestVersion string` and the `curr == nil` if block. Hence same goes for `fetched, err := manifest.FetchManifest()`.

WDYT?